libcst/matchers/_matcher_base.py

```diff
@@ -342,16 +484,41 @@ def ZeroOrOne(
     matcher: Union[_MatcherT, DoNotCareSentinel] = DoNotCareSentinel.DEFAULT
 ) -> AtMostN[Union[_MatcherT, DoNotCareSentinel]]:
     """
-    Used as a convenience wrapper to :class:`AtMostN```(n=1)``. This is effectively a
-    maybe clause.
+    Used as a convenience wrapper to :class:`AtMostN` when ``n`` is equal to ``1``.
+    This is effectively a maybe clause.
+
+    For example, this will match any function call with zero or one integer
+    argument::
+
+        m.Call(args=(m.ZeroOrOne(m.Arg(m.Integer())),))
+
+    This will match any function call that has two or three arguments, and
+    the first and last arguments are strings::
+
+        m.Call(args=(m.Arg(m.SimpleString()), m.ZeroOrOne(), m.Arg(m.SimpleString())))
+
     """
     return cast(AtMostN[Union[_MatcherT, DoNotCareSentinel]], AtMostN(matcher, n=1))
 
 
 def DoesNotMatch(obj: _OtherNodeT) -> _OtherNodeT:
```

Comment:
Question, this is analogy to `matches()`, why not call it `do_not_matches()`

Reply:
Its not the opposite of `matches()`. Its a predicate used inside matchers. That's why it isn't named opposite of `matches()`.